### PR TITLE
initial setup

### PR DESCRIPTION
**Pull request by davist11**
Some initial setup stuff and deleting some of the gatsby starter code. This first PR is a little gross, future ones will be better scoped.

### CODE REVIEW

File: gatsby-node.js [hunk 1]
```diff
@@ -4,84 +4,85 @@ const { createFilePath } = require('gatsby-source-filesystem')
...
+        // Tag pages:
+        let tags = []
+        // Iterate through each post, putting all found tags into `tags`
+        posts.forEach((edge) => {
+            if (_.get(edge, `node.frontmatter.tags`)) {
+                tags = tags.concat(edge.node.frontmatter.tags)
+            }
+        })
+        // Eliminate duplicate tags
+        tags = _.uniq(tags)
```
Comment (jeremyfrank) [inline gatsby-node.js:59]: Maybe make a mental note to remove all this tags code if you are not going to be doing any "tags" stuff for blog posts.

File: package.json [hunk 2]
```diff
@@ -41,10 +43,34 @@
...
+    "fix-css": "yarn lint-css --fix",
+    "fix-js": "yarn lint-js --fix",
+    "prettify": "yarn prettier --write",
+    "prettier": "prettier './**/*.{js,jsx,json,yml,yaml,css,less,scss,ts,tsx,md,mdx,graphql,vue}'",
+    "format": "yarn prettify && yarn fix-js && yarn fix-css",
     "test": "echo \"Error: no test specified\" && exit 1"
   },
   "devDependencies": {
-    "prettier": "^2.0.5"
+    "@fullhuman/postcss-purgecss": "^2.3.0",
```
Comment (jeremyfrank) [inline package.json:56]: Might want to look at `gatsby-plugin-purgecss` instead of this one from fedpack.

File: src/css/reset/index.css [hunk 3]
```diff
@@ -0,0 +1,287 @@
+a,
+abbr,
+address,
+area,
```
Comment (jeremyfrank) [inline src/css/reset/index.css:4]: Maybe double check to see if Tailwind's normalize + light reset approach suits your needs.

File: static/admin/config.yml [hunk 4]
```diff
@@ -1,66 +1,227 @@
...
+   "backend": {
+      "name": "git-gateway",
+      "branch": "master",
+      "commit_messages": {
+         "create": "Create {{collection}} “{{slug}}”",
+         "update": "Update {{collection}} “{{slug}}”",
+         "delete": "Delete {{collection}} “{{slug}}”",
+         "uploadMedia": "[skip ci] Upload “{{path}}”",
+         "deleteMedia": "[skip ci] Delete “{{path}}”"
+      }
```
Comment (jeremyfrank) [inline static/admin/config.yml:11]: Might not need to include these, unless you want to customize them. These are the default settings.

File: package.json (hunk 2 above)
Comment (davist11) [reply package.json:56]: I _think_ I can actually delete these, since I'm using the tailwind purge.

Comment (jeremyfrank) [reply package.json:56]: Oh, that's right, Tailwind has purge included now. I forgot about that.

File: static/admin/config.yml (hunk 4 above)
Comment (davist11) [reply static/admin/config.yml:11]: Is this documented somewhere?

Comment (jeremyfrank) [reply static/admin/config.yml:11]: https://www.netlifycms.org/docs/beta-features/#commit-message-templates

File: src/css/global.css [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+/**
+* This injects any component classes registered by plugins.
+*/
+@import 'tailwindcss/components';
+
+/**
+* This injects any component classes that have been extracted
+*/
+/*! purgecss start ignore */
+@import 'components/*';
```
Comment (leobauza) [inline src/css/global.css:25]: we don't do components unless we absolutely have to 💪 ...so comment out or delete?

File: src/css/base/core.css [hunk 6]
```diff
@@ -0,0 +1,4 @@
+body {
+    @apply bg-white;
+    @apply font-sans;
+}
```
Comment (leobauza) [inline src/css/base/core.css:4]: feels like these files are broken up more than they need to be? (i mean css rest and base stuff)

File: src/components/Header.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    },
+]
+
+const Header = () => (
+    <header className={cx('bg-black text-white')}>
+        <h1>Beacon Biosignals</h1>
+        <nav>
+            <ul>
+                {links.map(({ name, link }, index) => (
+                    <li key={index}>
```
Comment (leobauza) [inline src/components/Header.js:26]: not important but... `link` is unique so you don't need the index (also using the index is sometimes fine and sometimes the worst u can do so always avoid it if possible) https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318

File: src/components/Layout.js [hunk 8]
```diff
@@ -1,58 +1,62 @@
...
-    </div>
-  )
+                <meta property="og:type" content="business.business" />
+                <meta property="og:title" content={title} />
+                <meta property="og:url" content="/" />
+                <meta
+                    property="og:image"
+                    content={`${withPrefix('/')}img/og-image.jpg`}
+                />
+            </Helmet>
```
Comment (leobauza) [inline src/components/Layout.js:50]: pretty sure something in gatsby config can do all the icon stuff for you or most of this (I think?) is this part of the starter?

File: src/pages/contact/index.js [hunk 9]
```diff
@@ -3,113 +3,129 @@ import { navigate } from 'gatsby-link'
...
 
 export default class Index extends React.Component {
-  constructor(props) {
-    super(props)
-    this.state = { isValidated: false }
-  }
+    constructor(props) {
+        super(props)
+        this.state = { isValidated: false }
+    }
```
Comment (leobauza) [inline src/pages/contact/index.js:18]: why isn't this a functional component with hooks?

File: src/templates/what-you-get-page.js [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+                <div className="columns">
+                    <div className="column is-10 is-offset-1">
+                        <div className="section">
+                            <h2 className="title is-size-3 has-text-weight-bold is-bold-light">
+                                {title}
+                            </h2>
+                            <PageContent
+                                className="content"
+                                content={content}
+                            />
```
Comment (leobauza) [inline src/templates/what-you-get-page.js:26]: Kinda confused by what is happening here...why isn't this being handled with `children` somehow?

File: gatsby-browser.js [hunk 11]
```diff
@@ -0,0 +1 @@
+import './src/css/global.css'
```
[the remark this answers is not in the document]
Comment (davist11) [reply gatsby-browser.js:1]: Picked that up from @jeremyfrank

File: src/components/Header.js (hunk 7 above)
Comment (davist11) [reply src/components/Header.js:26]: Interesting, pretty sure I was getting linter complaints for not using the key

File: src/components/Layout.js (hunk 8 above)
Comment (davist11) [reply src/components/Layout.js:50]: Yeah part of the starter

File: src/pages/contact/index.js (hunk 9 above)
Comment (davist11) [reply src/pages/contact/index.js:18]: This was all part of the starter as well. You have a resource you can point me to for what it should be?

File: src/templates/what-you-get-page.js (hunk 10 above)
Comment (davist11) [reply src/templates/what-you-get-page.js:26]: Hmm, this is how they set it up in the starter (I just copied and pasted from that to create a new page template)

File: postcss.config.js [hunk 12]
```diff
@@ -0,0 +1,3 @@
+module.exports = () => ({
+    plugins: [require('postcss-easy-import'), require('tailwindcss')],
```
[the remark this answers is not in the document]
Comment (davist11) [reply postcss.config.js:2]: from my limited testing, that purge plugin seems to work just fine in dev

File: src/components/Header.js (hunk 7 above)
Comment (leobauza) [reply src/components/Header.js:26]: no you 100% need a key...I just mean don't use the `index` as the key use something unique that isn't the `index`

File: src/pages/contact/index.js (hunk 9 above)
Comment (leobauza) [reply src/pages/contact/index.js:18]: So for the types of components see: https://reactjs.org/docs/components-and-props.html#function-and-class-components

Most of the time it's easier/better/nicer to use functional components with hooks (there are reasons to use class components but...I don't ever run into those reasons and couldn't articulate them well)

```jsx
const MyComponent = () => {
   // useState hook for the state
  const [isValidated, setIsValidated] = React.useState(false)

  // ...etc

  return (
    <div>my component goes here</div>
  )
}
```

File: src/templates/what-you-get-page.js (hunk 10 above)
Comment (leobauza) [reply src/templates/what-you-get-page.js:26]: gotcha I guess I dunno everything it is doing just seems like a weird pattern...I may pull this and see and then get a better idea

File: .eslintrc [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+        "jest/globals": true
+    },
+    "parser": "babel-eslint",
+    "plugins": ["prettier", "react", "react-hooks", "jest"],
+    "extends": [
+        "eslint:recommended",
+        "plugin:prettier/recommended",
+        "plugin:react/recommended",
+        "plugin:jest/recommended"
+    ],
```
[the remark this answers is not in the document]
Comment (davist11) [reply .eslintrc:15]: I probably don't even need this jest stuff...it's a 3 page site

**Pull request merged**